tools/autograd/gen_variable_type.py

```diff
@@ -331,14 +403,30 @@ def unpack_args():
         body.extend(unpack_args())
 
         i = 0
+        added_derivative_tensor = False
+        added_derivative_tensorlist = False
         for arg in op['python_arguments']:
             derivative = arg.get('derivative')
             if derivative is None:
                 continue
-            body.append(DERIVATIVE.substitute({
-                'i': i,
-                'derivative': derivative,
-            }))
+
+            if arg['type'] == 'TensorList':
+                error_msg = "derivatives don't support specifying both a TensorList and non-TensorList derivative yet"
```

Comment:
uh let's inline this. it's confusing when it's not in the error path.

I now see that it's used in two places, but I'd still inline it, or I'd put an assertion outside of this for loop

Reply:
BTW what's the problem with having both? they have different templates?

Reply:
Our DERIVATIVE_TENSORLIST template isn't smart enough yet; consider if you had a TensorList "tl" and Tensor "t" argument in that order.  The return value of the tl backward function needs to be assigned to the first tl.size() grad inputs (missing nice python slice syntax).  So instead we just assign the return to the entire grad_inputs variable_list.  Since this is never actually used (at least yet), punting made the most sense.